corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -449,6 +488,7 @@ hqDefine("data_dictionary/js/data_dictionary", [
                 prop.deprecated.subscribe(changeSaveButton);
                 prop.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
                 prop.allowedValues.on('change', changeSaveButton);
+                prop.hasChanges = true;
```

Comment:
Differences between this code block, and the similar code block near line 60:

1. `propertyListItem.hasChanges = true` is set here, but not near line 60.
2. Near line 60 we call `propertyListItem.allowedValues.on('change', propObj.allowedValuesChanged)` but not here.
3. Near line 60 we call `subscribePropObservable()`.

Are these differences intentional? Can this duplicated code be moved and deduplicated? Should the differences (1) and/or (2) also be moved and combined, or should they stay where they are and remain different?

Reply:
More context: Does this prefactor for chunked loading look right? [d2827d5462e](https://github.com/dimagi/commcare-hq/commit/d2827d5462ea9f0b6409294eab52de6eb8161aa7#diff-8b20f9142ae4109168ec759806fe51ff14daaa7eec1f7f18e04c08f0d247e903R258)

Reply:
@kaapstorm Regarding (1), we set `hasChanges` here since it is a new case property and so it should automatically be marked for save. This differs from case properties around line 60 which already exist and so don't yet have any new changes to save. 

The same is true for (2) and (3). Since the new case property is automatically marked as needing to be saved, there is no need for us to track any changes to the case property as `hasChanges` will remain `true` until the new case property is saved.

> More context: Does this prefactor for chunked loading look right? [d2827d5462e](https://github.com/dimagi/commcare-hq/commit/d2827d5462ea9f0b6409294eab52de6eb8161aa7#diff-8b20f9142ae4109168ec759806fe51ff14daaa7eec1f7f18e04c08f0d247e903R258)

The prefactor looks right to me. It does change the logic slightly by tracking case property changes for both new and existing case properties. You could potentially throw all the subscribe hooks into a function that only gets called for existing case properties. But I think it's also fine to leave as is to keep things more consistent. I'll leave it up to your judgement on which approach to take for the prefactor.